Mage/src/main/java/mage/abilities/common/PutCounterOnCreatureTriggeredAbility.java

```diff
@@ -0,0 +1,90 @@
+package mage.abilities.common;
+
+import mage.abilities.TriggeredAbilityImpl;
+import mage.abilities.effects.Effect;
+import mage.constants.Zone;
+import mage.counters.Counter;
+import mage.filter.FilterPermanent;
+import mage.filter.common.FilterCreaturePermanent;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+import mage.target.targetpointer.FixedTarget;
+
+/**
+ * "Whenever you put one or more counters on a creature " triggered ability
+ *
+ * @author PurpleCrowbar
+ */
+public class PutCounterOnCreatureTriggeredAbility extends TriggeredAbilityImpl {
+
+    private final Counter counterType; // when null, any counter type is accepted
+    protected FilterPermanent filter;
+    private final boolean setTargetPointer;
+
+    public PutCounterOnCreatureTriggeredAbility(Effect effect) {
+        this(effect, false);
+    }
+
+    public PutCounterOnCreatureTriggeredAbility(Effect effect, boolean optional) {
+        this(effect, null, new FilterCreaturePermanent(), optional, false);
```

Comment:
The design pattern I've seen for these would have this constructor call `this(effect, null, optional)`. That way each of them cascade into the next one up so that defaults only have to be set once.